src/gui/ScintillaEditor.cc

```diff
@@ -1201,11 +1200,11 @@ bool ScintillaEditor::modifyNumber(int key)
   qsci->SendScintilla(QsciScintilla::SCI_SETEMPTYSELECTION);
   qsci->setCursorPosition(line, index);
 
-  auto begin = QRegExp(R"([-+]?\d*\.?\d*$)").indexIn(text.left(index));
+  auto begin = text.left(index).indexOf(QRegularExpression(R"([-+]?\d*\.?\d*$)"));
 
-  QRegExp rx("[_a-zA-Z]");
+  QRegularExpression rx(QRegularExpression::anchoredPattern(QString("[_a-zA-Z]")));
   auto check = text.mid(begin - 1, 1);
-  if (rx.exactMatch(check)) return false;
+  if (rx.match(check).hasMatch()) return false;
```

Comment:
Not sure this will work; exact matching semantics from QRegExp isn't the same as regular matching in QRegularExpression. I didn't read the porting guide, but it's here: https://doc.qt.io/qt-6/qregexp.html#porting-to-qregularexpression

Reply:
Ah, anchoredPattern. Thanks for the clarification!